src/distributions.jl

```diff
@@ -22,12 +22,21 @@ function default_range(m::Distributions.MixtureModel, alpha = 0.0001)
 end
 
 yz_args(dist) = default_range(dist)
-yz_args(dist::Distribution{N, T}) where N where T<:Discrete = (UnitRange(Int.(default_range(dist))...),)
+function yz_args(dist::DiscreteUnivariateDistribution)
+    minval, maxval = extrema(dist)
+    if isfinite(minval) && isfinite(maxval)  # bounded
+        sup = support(dist)
+        return sup isa AbstractVector ? (sup,) : (reduce(vcat, sup),)
+    else  # unbounded
+        return (UnitRange(default_range(dist)...),)
+    end
+end
 
 # this "user recipe" adds a default x vector based on the distribution's μ and σ
 @recipe function f(dist::Distribution)
     if dist isa Distribution{Univariate,Discrete}
-        seriestype --> :scatterpath
+        seriestype --> :sticks
+        markershape --> :circle
```

Comment:
Might be better to use some logic here to only set the `markershape` if `seriestype` hasn't been set. Otherwise the user asks for a `path` and we coerce it to show scatter points.